pkg/ipcache/ipcache.go

```diff
@@ -613,6 +621,7 @@ func (ipc *IPCache) deleteLocked(ip string, source source.Source) (namedPortsCha
 			scopedLog.Debug("Deleting CIDR shadowed by endpoint IP")
 			callbackListeners = false
 		}
+		ipc.prefixLengths.Delete([]netip.Prefix{cidrCluster.AsPrefix()})
```

Comment:
Similar to above: Why not extract the prefixes here and only delete it from `prefixLengths` below line 647.